semanticdb/metai/src/main/scala/scala/meta/internal/metai/Main.scala

```diff
@@ -47,12 +47,11 @@ final class Main(settings: Settings, reporter: Reporter) {
   }
 
   private def processEntry(file: ClasspathFile): Boolean = {
-    var hasSemanticdb = false
+    val hasSemanticdb = file.path.resolve("META-INF").resolve("semanticdb").isDirectory
     val ls = FileIO.listAllFilesRecursively(file.path)
```

Comment:
If you can assume the directory zip entry `META-INF/semanticdb` exists then you can simplify this logic with

```scala
val root = file.path.resolve("META-INF").resolve("semanticdb")
val ls = FileIO.listAllFilesrecursively(root)
ls.files.foreach { ... }
```